libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -526,31 +544,30 @@ - (void)resumeSyncUpOneEntry:(SFSyncState*)sync
         completeBlockUpdate(d);
     };
     
-    NSArray *fieldList = sync.options.fieldlist;
-    
-    // Getting type and id
-    NSString* objectType = [SFJsonUtils projectIntoJson:record path:kObjectTypeField];
-    NSString* objectId = record[target.idFieldName];
-    
-    // Fields to save (in the case of create or update)
-    NSMutableDictionary* fields = [NSMutableDictionary dictionary];
-    if (action == SFSyncUpTargetActionCreate || action == SFSyncUpTargetActionUpdate) {
-        for (NSString *fieldName in fieldList) {
-            if (![fieldName isEqualToString:target.idFieldName] && ![fieldName isEqualToString:target.modificationDateFieldName]) {
-                NSObject* fieldValue = [SFJsonUtils projectIntoJson:record path:fieldName];
-                if (fieldValue != nil)
-                    fields[fieldName] = fieldValue;
+    // Update failure handler
+    SFSyncUpTargetErrorBlock failBlockUpdate = ^ (NSError* err){
```

Comment:
When we get a 404 during an update, we don't fail anymore
We either recreate the record (if merge mode is overwrite)
Or we leave the local record unchanged (if merge mode is leave if changed)